File: project4/Go/task2.go
/*
  CS333 Project 4
  Part II: Selected Language
  Go: Task 2
  Shows how functions are treated in Go as a data type

  Yixuan Qiu
  Mar 8 2020
*/

package main

import (
	"fmt"
	"math"
)

// square function that takes a function as an argument
func square(sqrFunc func(num float64) float64) {
	fmt.Println("The square of 3 is:", sqrFunc(3))
}

func main() {
	// assign a function to a variable
	squareOfNum := func(num float64) float64 {
		result := math.Pow(num, 2)
		return result
	}

	// pass function as an argumnt in square function
	square(squareOfNum)
}

File: project4/task2.js
/**
 * CS333 Project 4
 * task2.js
 * Program that shows how functions are treated in JS
 * 
 * Di Luo
 * 03/11/2020
 */

function func(){
    console.log("func executes");
}

console.log(func instanceof Object);

var f = func();
f;

function func2(){
    console.log("Testing func in func.");
    func();
}

func2();

// function (){
//     console.log("Function without name");
// }

var f2 = function(){
    console.log("Function expression");
}
f2();